pydicom/framereader.py

```diff
@@ -0,0 +1,1182 @@
+
+    fp.seek(first_frame_location, 0)
+    return basic_offset_table
+
+
+def get_dataset_copy_with_frame_attrs(
+        original_dataset: Dataset,
+        optional_attributes: Tuple[str, ...] = _OPTIONAL_DATASET_ATTRIBUTES
+) -> Dataset:
+    """Create a copy of original_dataset with only the data elements needed
+    to parse frames
+
+    Parameters
+    ----------
+    original_dataset: Dataset
+        the dataset for which to create a minimal copy
+    optional_attributes: Tuple[str]
+
+    Returns
+    -------
+    Dataset
+        the copy of original_dataset
+
+    Raises
+    ------
+    ValueError
+        When original_dataset is missing attributes required for frame reading
+    """
+    ds_copy = Dataset()
+    ds_copy.file_meta = FileMetaDataset()
+    missing_attributes = list()
+    ds_copy.is_little_endian = original_dataset.is_little_endian
+    ds_copy.is_implicit_VR = original_dataset.is_implicit_VR
+    if not hasattr(original_dataset, "file_meta"):
+        transfer_syntax = None
+    else:
+        transfer_syntax = getattr(
+            original_dataset.file_meta, "TransferSyntaxUID", None
+        )
+    if transfer_syntax is None:
+        missing_attributes.append("file_meta.TransferSyntaxUID")
+    else:
+        ds_copy.file_meta.TransferSyntaxUID = transfer_syntax
+
+    for r_attribute in _REQUIRED_DATASET_ATTRIBUTES:
+        if not hasattr(original_dataset, r_attribute):
+            missing_attributes.append(r_attribute)
+        if missing_attributes:
+            continue
+        original_value = getattr(original_dataset, r_attribute)
+        setattr(ds_copy, r_attribute, original_value)
+
+    if missing_attributes:
+        del ds_copy
+        raise ValueError(
+            "Cannot make a copy of original_dataset as is missing the "
+            "following attributes required for frame reading: "
+            f"{missing_attributes}"
+        )
+    # not required, but need to copy if they exist
+    for attr in optional_attributes:
+        attr_value = original_dataset.get(attr)
+        setattr(ds_copy, attr, attr_value)
+    return ds_copy
+
+
+def decode_frame(
+        frame_bytes: bytes, original_dataset: Dataset
+) -> "numpy.ndarray":
+    """Decode the frame_bytes provided from original_dataset as a numpy ndarray
+
+    Parameters
+    ----------
+    frame_bytes: bytes
+        the bytes corresponding to a frame within DICOM PixelData
+    original_dataset: Dataset
+        the original_dataset representing the file from which frame_bytes were
+        retrieved
+
+    Returns
+    -------
+    numpy.ndarray
+        array representation of frame_bytes
+    """
+    ds_temp = get_dataset_copy_with_frame_attrs(original_dataset)
+    if ds_temp.file_meta.TransferSyntaxUID.is_encapsulated:
+        ds_temp.PixelData = encapsulate(frames=[frame_bytes])
+    else:
+        ds_temp.PixelData = frame_bytes
+
+    return ds_temp.pixel_array
+
+
+class BasicOffsetTable(List):
```

Comment:
I would not make that a subclass of `List` and then add `to_dict()` and `from_dict()` methods. If you really want to implement it as a class, I suggest using a `NamedTuple` instead.